vcd/resource_vcd_org_vdc_template.go

```diff
@@ -0,0 +1,772 @@
+					GatewayInterfaces: &types.GatewayInterfaces{GatewayInterface: []*types.GatewayInterface{
+						{
+							Name:        edgeGatewayBindingId,
+							DisplayName: edgeGatewayBindingId,
+							Connected:   true,
+							Network: &types.Reference{
+								HREF: edgeGatewayBindingId,
+							},
+							QuickAddAllocatedIpCount: gatewayBlock["ip_allocation_count"].(int),
+						},
+					}},
+				},
+			},
+			Network: &types.OrgVDCNetwork{
+				Name:        gatewayBlock["network_name"].(string),
+				Description: gatewayBlock["network_description"].(string),
+				Configuration: &types.NetworkConfiguration{
+					IPScopes: &types.IPScopes{IPScope: []*types.IPScope{
+						{
+							Gateway:               ip.String(),
+							Netmask:               fmt.Sprintf("%d.%d.%d.%d", cidr.Mask[0], cidr.Mask[1], cidr.Mask[2], cidr.Mask[3]),
+							SubnetPrefixLengthInt: &prefixLength,
+							IPRanges:              &types.IPRanges{IPRange: staticPools},
+						},
+					}},
+					FenceMode: "natRouted",
+				},
+			},
+		}
+		if gatewayEdgeClusterSet {
+			gateway.Gateway.Configuration.EdgeClusterConfiguration = &types.EdgeClusterConfiguration{PrimaryEdgeCluster: &types.Reference{HREF: edgeGatewayBindingId}}
+		}
+	}
+
+	// Storage profiles
+	spBlocks := d.Get("storage_profile").(*schema.Set).List()
+	storageProfiles := make([]*types.VdcStorageProfile, len(spBlocks))
+	for i, sp := range spBlocks {
+		spBlock := sp.(map[string]interface{})
+		storageProfile := &types.VdcStorageProfile{
+			Name:    spBlock["name"].(string),
+			Enabled: addrOf(true),
+			Units:   "MB",
+			Limit:   int64(spBlock["limit"].(int)),
+			Default: spBlock["default"].(bool),
+		}
+		storageProfiles[i] = storageProfile
+	}
+
+	allocationModel := getVdcTemplateType(d.Get("allocation_model").(string))
+	settings := types.VMWVdcTemplate{
+		NetworkBackingType:   "NSX_T", // The only supported network provider
+		ProviderVdcReference: pvdcs,
+		Name:                 d.Get("name").(string),
+		Description:          d.Get("description").(string),
+		TenantName:           d.Get("tenant_name").(string),
+		TenantDescription:    d.Get("tenant_description").(string),
+		VdcTemplateSpecification: &types.VMWVdcTemplateSpecification{
+			Type:                    allocationModel,
+			NicQuota:                d.Get("nic_quota").(int),
+			VmQuota:                 d.Get("vm_quota").(int),
+			ProvisionedNetworkQuota: d.Get("provisioned_network_quota").(int),
+			GatewayConfiguration:    gateway,
+			StorageProfile:          storageProfiles,
+			ThinProvision:           d.Get("enable_thin_provisioning").(bool),
+			FastProvisioningEnabled: d.Get("enable_fast_provisioning").(bool),
+		},
+	}
+
+	// Get the compute configuration. There are too many combinations to perform "smart validations" here,
+	// and given that VCD behaves well when input is incorrect, we skip them.
+	// Schema guarantees that there's exactly 1 item, so we can retrieve every field it directly.
```

Comment:
```suggestion
	// Schema guarantees that there's exactly 1 item, so we can retrieve every field directly.
```

Reply:
Fixed! I did a last minute change of that comment and ended with a leftover.